src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -713,6 +713,11 @@ internal void SetRequestHeaders()
                         ThrowOOMIfFalse(Interop.Http.SListAppend(slist, NoContentType));
                     }
                 }
+                else
+                {
+                    // Remove the Content-Type header libcurl adds by default.
+                    ThrowOOMIfFalse(Interop.Http.SListAppend(slist, NoContentType));
+                }
```

Comment:
This is ok, but it'd be nice if possible to consolidate the logic of actually adding the header, e.g. something like:
```C#
bool suppressContentType = true;
if (_requestMessage.Content != null)
{
    AddRequestHeaders(_requestMessage.Content.Headers, slist);
    suppressContentType = (_requestMessage.Content.Headers.ContentType == null);
}

if (suppressContentType)
{
    // Remove the Content-Type header libcurl adds by default)
    ThrowOOMIfFalse(Interop.Http.SListAppend(slist, NoContentType);
}
```